bitcoin/src/merkle_tree/mod.rs

```diff
@@ -129,8 +159,83 @@ mod tests {
             hashes_array[i] = hash;
         }
 
-        let from_iter = calculate_root(hashes_iter);
+        let from_iter = {
+            let before = Instant::now();
+            let from_iter = calculate_root(hashes_iter);
+            let delta = (Instant::now() - before).as_secs_f64();
+            println!("calculate_root: {delta} s");
+            from_iter
+        };
```

Comment:
This is noisy, you should use `#[bench]` instead and make sure to correctly use `black_box` among other things.

Reply:
AFAIK, #[bench] is unstable and it will be phase out

```
error: use of unstable library feature 'test': `bench` is a part of custom test frameworks which are unstable
   --> bitcoin\src\merkle_tree\mod.rs:213:7
    |
213 |     #[bench]
    |       ^^^^^
    |
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
    = note: for more information, see issue #64266 <https://github.com/rust-lang/rust/issues/64266>
    = note: `#[deny(soft_unstable)]` on by default
```

Reply:
Just use it on nightly, we already have such benches, no big issue. Though `black_box` is now stable so perhaps `criterion` could work but that's another dependency. IDK if it's worse to be broken by nightly or a dep but given that `#[bench]` didn't change for a while and the devs don't have much time to mess with it, it should be fine. :)

Reply:
I'd like to move to criterion but that is not a debate for this PR :). Agreed, just use `#[bench]` for now.